GiniVision/Classes/AnalysisContainerViewController.swift

```diff
@@ -64,7 +66,11 @@ internal class AnalysisContainerViewController: UIViewController, ContainerViewC
         let delegate = (navigationController as? GiniNavigationViewController)?.giniDelegate
         delegate?.didCancelAnalysis?()
         
-        _ = navigationController?.popViewController(animated: true)
+        if self == navigationController?.topViewController {
```

Comment:
Another place where you manipulate the navigation stack in a fragile way. Didn't we get rid of those?
And I really don't understand what's happening here.

Reply:
We didn't for the Screen API, since we still use the containers. You cannot pop a view controller inside a navigation controller if it is the rootViewController, that is way I need to check if the last one.
But this is wrong, it should be the rootViewController not the topViewController.